### ISSUE TITLE
Response never ends with large body

### ISSUE THREAD

**Issue by daiheitan** [hyperium/hyper#1032]
```Rust
Response::new()
    .with_header(header::ContentType::jpeg())
    .with_header(header::ContentLength(buf.len() as u64))
    .with_body(buf)
```

If the `buf` of `Vec<u8>` has a length too large (like 60000), hyper will never finish the response. If I shut down the server while browser is waiting for the (endless) response, this is a content length mismatch error in browser.

This happens on CentOS, not re-producing on OSX

**Comment by daiheitan** [comment]
closed by  cfcbd8c

**Comment by seanmonstar** [comment]
For history: the issue was that hyper noticed a partial write, and simply waited thinking that the socket was blocked. However, since the write returned `Ok(n)`, tokio did not know to re-register the socket with epoll for the write event, and so we sat around waiting forever. The solution was to loop on the write until it returned `EWOULDBLOCK`, which means tokio would then re-register interest.
